Sources/NIOPosix/PendingWritesManager.swift

```diff
@@ -463,16 +498,82 @@ final class PendingStreamWritesManager: PendingWritesManager {
         return self.didWrite(itemCount: result.itemCount, result: result.writeResult)
     }
 
-    /// Fail all the outstanding writes. This is useful if for example the `Channel` is closed.
+    /// Fail all the outstanding writes.
     func failAll(error: Error, close: Bool) {
         if close {
             assert(self.isOpen)
             self.isOpen = false
+            self.state.removeAll()?.fail(error)
```

Comment:
Not sure if it's just the diff but it looks like this function also failed all the promises if `!close`, that doesn't seem to happen anymore.